lib/timers.js

```diff
@@ -477,47 +477,109 @@ exports.clearImmediate = function(immediate) {
 
 var unrefList, unrefTimer;
 
+function _makeTimerTimeout(timer) {
+  var domain = timer.domain;
+  var msecs = timer._idleTimeout;
+
+  // Timer has been unenrolled by another timer that fired at the same time,
+  // so don't make it timeout.
+  if (!msecs || msecs < 0)
+    return;
+
+  if (!timer._onTimeout)
+    return;
+
+  if (domain) {
+    if (domain._disposed)
+      return;
+
+    domain.enter();
+  }
+
+  debug('unreftimer firing timeout');
+  L.remove(timer);
+  timer._called = true;
+  _runOnTimeout(timer);
+
+  if (domain)
+    domain.exit();
+}
+
+function _runOnTimeout(timer) {
+  try {
+    var threw = true;
+    timer._onTimeout();
+    threw = false;
+  } finally {
+    if (threw) process.nextTick(unrefTimeout);
```

Comment:
is it possible that in the same execution phase multiple timers could be called that throw? or are timers no longer processed after one has throw?

If the former then would it be useful to make sure `unrefTimeout` is scheduled only once? Otherwise it may be queued to run several times.